cnf-certification-test/accesscontrol/suite.go

```diff
@@ -32,111 +32,119 @@ import (
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestContainerHostPort)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Containers)
-		TestContainerHostPort(&env)
-	})
-	// container security context: host network
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostNetwork)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodHostNetwork(&env)
-	})
-	// pod host path
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPath)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodHostPath(&env)
-	})
-	// pod host ipc
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostIPC)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodHostIPC(&env)
-	})
-	// pod host pid
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPID)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodHostPID(&env)
-	})
-	// Namespace
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestNamespaceBestPracticesIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Namespaces)
-		TestNamespace(&env)
-	})
-	// pod service account
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodServiceAccountBestPracticesIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodServiceAccount(&env)
-	})
-	// pod role bindings
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodRoleBindingsBestPracticesIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodRoleBindings(&env)
-	})
-	// pod cluster role bindings
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodClusterRoleBindingsBestPracticesIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestPodClusterRoleBindings(&env)
-	})
-	// automount service token
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodAutomountServiceAccountIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Pods)
-		TestAutomountServiceToken(&env)
-	})
-	// one process per container
-	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestOneProcessPerContainerIdentifier)
-	ginkgo.It(testID, ginkgo.Label(testID), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Containers)
-		TestOneProcessPerContainer(&env)
-	})
+	for _, t := range testContainer {
+		tid := identifiers.XformToGinkgoItIdentifier(t.id)
+		ginkgo.It(tid, ginkgo.Label(tid), func() {
+			testhelper.SkipIfEmptyAny(ginkgo.Skip, env.Containers)
```

Comment:
To be honest, I prefer the current pattern. We are loosing the ability to have a custom skip function per test case ( the testhelper.SkipIfEmptyAny functions). It works in this case but not all suites use the same. Also, modifying the common code will need regression test for all tests. I feel this part should not be optimized.

Reply:
I agree that it's more uniform across all of the test suites to keep it the way it is.

Reply:
To have the ability I can add skip function to the Test structure. The current function is too long, the actions are out of order and looks messy.

I would add Containers/Pods/Namespaces to the structure as well and have only one cycle, but I did not find a way to do in Go, it will be a pointer on a data member in C++, which is to complex for Go :-)

Reply:
If we are agree on this I will fix the other suites as well.